files/image_config/hostcfgd/hostcfgd

```diff
@@ -219,6 +220,537 @@ class AaaCfg(object):
+        try:
+            ver = "-4"
+            mgmt_conf = netaddr.IPNetwork(ip_prefix)
+            if mgmt_conf.version == 6:
+                ver = "-6"
+            cmd = "ip {} addr {} {} dev {}".format(ver, op.lower(), ip_prefix, ifname)
+            syslog.syslog(syslog.LOG_INFO, "Configure ip_prefix, op : {}, ip_prefix: {}, cmd : {}, mgmt_conf: {}".format(op, ip_prefix, cmd, mgmt_conf))
+            subprocess.check_call(cmd, shell=True)
+            cmd = "ip {} rule {} from {} table default pref 1003".format(ver, op.lower(), str(mgmt_conf.ip))
+            subprocess.check_call(cmd, shell=True)
+        except subprocess.CalledProcessError as err:
+            syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(err.cmd, err.returncode, err.output))
+            pass
+
+    def handle_mgmt_port_cfg(self, key, data):
+        mgmt_intf =  self.app_db.get_entry(APP_MGMT_PORT_TABLE, key)
+        is_all = False
+        if mgmt_intf == {}:
+            is_all = True
+
+        try:
+            if (('mtu' in data) and ((is_all == True) or (mgmt_intf == {}) or ('mtu' not in mgmt_intf) or (data['mtu'] != mgmt_intf['mtu']))):
+                self.mgmt_intf_mtu_set(key, data['mtu'], mgmt_intf)
+
+            if (('description' in data) and ((is_all == True) or (mgmt_intf == {}) or ('description' not in mgmt_intf) or (data['description'] != mgmt_intf['description']))):
+                self.mgmt_intf_description_set(key, data['description'], mgmt_intf)
+
+            if (('admin_status' in data) and ((is_all == True) or (mgmt_intf == {}) or ('admin_status' not in mgmt_intf) or (data['admin_status'] != mgmt_intf['admin_status']))):
+                self.mgmt_intf_admin_status_set(key, data['admin_status'], mgmt_intf)
+
+            if (('autoneg' in data) and ((is_all == True) or (mgmt_intf == {}) or ('autoneg' not in mgmt_intf) or (data['autoneg'] != mgmt_intf['autoneg']))):
+                self.mgmt_intf_autoneg_set(key, data['autoneg'], mgmt_intf)
+
+            if (('speed' in data) and ((is_all == True) or (mgmt_intf == {}) or ('speed' not in mgmt_intf) or (data['speed'] != mgmt_intf['speed']))):
+                autoneg = "on"
+                if 'autoneg' in data:
+                    if data['autoneg'] == "false":
+                        autoneg = "off"
+                self.mgmt_intf_speed_set(key, data['speed'], autoneg, mgmt_intf)
+        except:
+            syslog.syslog(syslog.LOG_ERR, "{} MGMT Port Config set failed.".format(key))
+
+        self.app_db.set_entry(APP_MGMT_PORT_TABLE, key, mgmt_intf)
+
+
+    def handle_mgmt_intf_cfg(self, key, data):
+        cur_appdb_key = key.replace('|', ':')
+        intf_keys = key.split("|")
+        cfgdb_entry = self.config_db.get_entry(CFG_MGMT_INTF_TABLE, key)
+        appdb_entry = self.app_db.get_entry(APP_MGMT_INTF_TABLE, cur_appdb_key)
+        op = "DELETE"
+        keys = self.config_db.get_keys(CFG_MGMT_INTF_TABLE)
+
+        for it in keys:
```

Comment:
Please provide a comment as to what is intended in this `for `section

Reply:
conect db apis' get_keys return list of keys can be tuple format or list  of string format.  to identify its create or update operation looping through all the keys, convert into a string format and compare.
Any way simplified the logic. Addressed the comments.